modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/filter/XPath.java

```diff
@@ -276,11 +276,8 @@ public Attribute set(
                                             new GeometryTypeImpl(
                                                     targetNodeType.getName(),
                                                     targetNodeType.getBinding(),
-                                                    crs != null
-                                                            ? crs
-                                                            : ((GeometryDescriptor)
-                                                                            actualDescriptor)
-                                                                    .getCoordinateReferenceSystem(),
+                                                    ((GeometryDescriptor) actualDescriptor)
```

Comment:
Note: the CRS passed in the constructor of XPath is now only used as a back-up, when there is no CRS feature data. Otherwise, the CRS from the feature data will be used (see AttributeBuilder).